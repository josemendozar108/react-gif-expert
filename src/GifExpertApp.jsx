import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {

  const [categories, setCategories] = useState([]); 

  const onAddCategory = (newCategory) => {

    if (categories.includes(newCategory)) return; 
      // console.log(newCategory); 
    // Evitar usar el "push"
    // categories.push(newCategory); 
    setCategories([newCategory, ...categories]); 


  }

  // console.log(categories); 


  return (
    <>
      
        <h1> GifExpertApp </h1>

        <AddCategory 
        onNewCategory={ (value) => onAddCategory(value)}
        />
        
        {
            categories.map( (category) => (
                <GifGrid
                  key={ category }
                  category={category}/> 
              ))

        } 
        
    </>
  )
}
