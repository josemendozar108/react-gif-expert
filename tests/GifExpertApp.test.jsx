import { render, screen } from '@testing-library/react'
import { GifExpertApp } from '../src/GifExpertApp'



describe('Pruebas de <GifExpertApp /> ', () => {

    test('Debe ser match con el snapshot', () => { 

       const {container} = render(<GifExpertApp /> )
       expect(container).toMatchSnapshot(); 
        
     }); 

 })