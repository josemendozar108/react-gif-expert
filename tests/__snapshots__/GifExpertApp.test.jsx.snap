// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifExpertApp />  Debe ser match con el snapshot 1`] = `
<div>
  <h1>
     GifExpertApp 
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
